htlcswitch/switch.go

```diff
@@ -1170,6 +1194,40 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 	}
 }
 
+// CheckCircularForward checks whether a forward is circular (arrives and
+// departs on the same link) and returns a link error if the switch is
+// configured to disallow this behaviour.
+func CheckCircularForward(incoming, outgoing lnwire.ShortChannelID,
+	allowCircular bool, paymentHash lntypes.Hash) *LinkError {
+
+	// Check that the incoming link is different to the outgoing link.
+	// If they are equal, the htlc is part of a circular route which may
+	// be used to lock up our liquidity.
+	isCircularRoute := incoming.ToUint64() == outgoing.ToUint64()
```

Comment:
I think the `ToUint64()` is not required here as these structs are comparable and comparison should actually be faster too.